Cython/Utility/CommonStructures.c

```diff
@@ -89,3 +89,88 @@ static PyTypeObject *__Pyx_FetchCommonTypeFromSpec(PyObject *module, PyType_Spec
     goto done;
 }
 
+/////////////// RegisterABC.proto ///////////////
+
+static void __Pyx_RegisterCommonTypeWithAbc(PyObject *type, const char *abc_name); /* proto */
+
+/////////////// RegisterABC ///////////////
+//@requires: ModuleSetupCode.c::CriticalSections
+//@requires: ObjectHandling.c::PyObjectGetAttrStrNoError
+
+static int __Pyx__RegisterCommonTypeWithAbc(PyObject* cython_runtime, PyObject *type, const char *abc_name)
+{
+    int result = -1;
+    PyObject *abc_class = NULL;
+    PyObject *py_abc_name = PyUnicode_FromString(abc_name);
+    if (unlikely(!py_abc_name))
+        return -1;
+    abc_class = __Pyx_PyObject_GetAttrStrNoError(cython_runtime, py_abc_name);
```

Comment:
I need to have another look at this actually. I don't think GetAttr works well with critical sections so I may need to make it more robust or not bother